storage/netapp_e/na_eseries_storage_system.py

```diff
@@ -0,0 +1,319 @@
+    choices: ['present', 'absent']
+  controller_addresses:
+    required: true
+    description:
+    - The list addresses for the out-of-band management adapter or the agent host.
+  array_wwn:
+    required: false
+    description:
+    - The WWN of the array to manage. Only necessary if in-band managing multiple arrays on the same agent host.
+  array_password:
+    required: false
+    description:
+    - The management password of the array to manage, if set.
+  enable_trace:
+    required: false
+    default: false
+    description:
+    - Enable trace logging for SYMbol calls to the storage system.
+  meta_tags:
+    required: false
+    default: None
+    description:
+    - Optional meta tags to associate to this storage system
+author: Kevin Hulquest (@hulquest)
+'''
+
+EXAMPLES = '''
+---
+    - name:  Presence of storage system
+      na_eseries_storage_system:
+        ssid: "{{ item.key }}"
+        state: present
+        api_url: "{{ netapp_api_url }}"
+        api_username: "{{ netapp_api_username }}"
+        api_password: "{{ netapp_api_password }}"
+        validate_certs: "{{ netapp_api_validate_certs }}"
+        controller_addresses:
+          - "{{ item.value.address1 }}"
+          - "{{ item.value.address2 }}"
+      with_dict: "{{ storage_systems }}"
+      when: check_storage_system
+'''
+
+RETURN = '''
+msg: Storage system removed.
+msg: Storage system added.
+'''
+
+
+def request(url, data=None, headers=None, method='GET', use_proxy=True,
+            force=False, last_mod_time=None, timeout=10, validate_certs=True,
+            url_username=None, url_password=None, http_agent=None, force_basic_auth=True, ignore_errors=False):
+    try:
+        r = open_url(url=url, data=data, headers=headers, method=method, use_proxy=use_proxy,
+                     force=force, last_mod_time=last_mod_time, timeout=timeout, validate_certs=validate_certs,
+                     url_username=url_username, url_password=url_password, http_agent=http_agent,
+                     force_basic_auth=force_basic_auth)
+    except HTTPError:
+        err = get_exception()
+        r = err.fp
+
+    try:
+        raw_data = r.read()
+        if raw_data:
+            data = json.loads(raw_data)
+        else:
+            raw_data = None
+    except:
+        if ignore_errors:
+            pass
+        else:
+            raise Exception(raw_data)
+
+    resp_code = r.getcode()
+
+    if resp_code >= 400 and not ignore_errors:
+        raise Exception(resp_code, data)
+    else:
+        return resp_code, data
+
+
+def main():
+    argument_spec = basic_auth_argument_spec()
+    argument_spec.update(dict(
+        state=dict(required=True, choices=['present', 'absent']),
+        ssid=dict(required=True, type='str'),
+        controller_addresses=dict(type='list'),
+        array_wwn=dict(required=False, type='str'),
+        array_password=dict(required=False, type='str'),
```

Comment:
To avoid password appearing in logs you should set `no_log=True`

See http://docs.ansible.com/ansible/developing_modules.html